Fabric.Terminology.API/Scripts/Terminology-Intall-Utilities.psm1

```diff
@@ -0,0 +1,135 @@
+    [String]$discoveryServiceUrl
+    [PSCredential]$iisUserCredentials
+}
+function Import-Modules() {
+    # Dos Install Utilities
+    $dosInstallUtilities = Get-Childitem -Path ./**/DosInstallUtilities.psm1 -Recurse
+    if($dosInstallUtilities.length -eq 0) {
+        Write-Warning "could not find dos install utilities. Manually installing "
+        Install-Module DosInstallUtilities -Scope CurrentUser
+        Import-Module DosInstallUtilities -Force
+    } else {
+        Import-Module -Name $dosInstallUtilities.FullName -Force
+    }
+	
+	# Fabric Install Utilities
+	$fabricInstallUtilities = ".\Fabric-Install-Utilities.psm1"
+    if(!(Test-Path $fabricInstallUtilities -PathType Leaf)){
+        Invoke-WebRequest -Uri https://raw.githubusercontent.com/HealthCatalyst/InstallScripts/master/common/Fabric-Install-Utilities.psm1 -Headers @{"Cache-Control"="no-cache"} -OutFile $dosInstallUtilities
+    }
+    Import-Module -Name $fabricInstallUtilities -Force
+}
+
+function Get-TerminologyConfig{
+    param(
+        [string] $Path
+    )
+    
+    $installSettings = Get-InstallationSettings "terminology"
+
+    $iisUser = Read-Host "Press Enter to accept the default IIS App Pool User"
+    $userEnteredPassword = Read-Host "Enter the password for $iisUser" -AsSecureString
+    $credential = New-Object -TypeName "System.Management.Automation.PSCredential" -ArgumentList $iisUser, $userEnteredPassword
+    [System.Reflection.Assembly]::LoadWithPartialName("System.DirectoryServices.AccountManagement") | Out-Null
+    $ct = [System.DirectoryServices.AccountManagement.ContextType]::Domain
+    $pc = New-Object System.DirectoryServices.AccountManagement.PrincipalContext -ArgumentList $ct, $credential.GetNetworkCredential().Domain
+    $isValid = $pc.ValidateCredentials($credential.GetNetworkCredential().UserName, $credential.GetNetworkCredential().Password)
+    if (!$isValid) {
+        Write-Error "Incorrect credentials for $iisUser"
+        throw
+    }
+
+    $config = [TerminologyConfig]::new()
+    $config.iisUserCredentials = New-Object System.Management.Automation.PSCredential ($iisUser, $userEnteredPassword)
+    $config.appName = $installSettings.appName
+    $config.appPool = $installSettings.appPool
+    $config.siteName = $installSettings.siteName
+    if ([string]::IsNullOrWhiteSpace($installSettings.discoveryServiceUrl)) {
+        $config.discoveryServiceUrl = Read-Host "Please enter the discovery service Uri (eg. https://SERVER/discoveryservice/v1)"
+    } else {
+        $config.discoveryServiceUrl = $installSettings.discoveryServiceUrl
+    }
+    
+
+	return $config
+}
+
+function Update-AppSettings{
+    param(
+        [TerminologyConfig] $config
+    )
+
+    $appPath = Get-WebURL "IIS:\Sites\$($config.siteName)\$($config.appName)"
+    $config.applicationEndpoint = $appPath.ResponseUri
+
+    $appSettings = "$(Get-IISWebSitePath -WebSiteName $config.siteName)\$($config.appName)\appsettings.json"
```

Comment:
This requires the WebAdminModule to be imported - maybe consider attempting the import at the top of the module so you can surface the error in a easier to digest fashion?